### ISSUE TITLE
py_reader怎么使用？训练和测试一起

### ISSUE THREAD

**Issue by Exception-star** [PaddlePaddle/Paddle#19226]
- 版本、环境信息：
    1）PaddlePaddle版本：1.5.1
    2）CPU：cpu
- 问题描述：想训练玩一个epoch, 进行一次test，执行以下代码不报错，没有任何输出
`

import paddle.fluid as fluid
import paddle

def net(image):
    conv = fluid.layers.conv2d(image,
                               num_filters=16,
                               filter_size=3,
                               stride=1,
                               padding=1,
                               act='relu')
    fc = fluid.layers.fc(conv,
                         size=10,
                         act='softmax')
    return fc

`

def main():

    image = fluid.layers.data(name='image', shape=[1, 28, 28], dtype='float32')
    label = fluid.layers.data(name='label', shape=[1], dtype='int64')

    # py_reader for train
    py_reader = fluid.io.PyReader(feed_list=[image, label],
                                  capacity=2,
                                  iterable=False)
    batch_data_train = paddle.batch(paddle.reader.shuffle(paddle.dataset.mnist.train(), buf_size=512), batch_size=256)
    py_reader.decorate_sample_list_generator(batch_data_train)

    # py_reader2 for test
    py_reader2 = fluid.io.PyReader(feed_list=[image, label],
                                   capacity=2,
                                   iterable=False)
    batch_data_test = paddle.batch(paddle.reader.shuffle(paddle.dataset.mnist.test(), buf_size=512), batch_size=256)
    py_reader2.decorate_sample_list_generator(batch_data_test)

    pred = net(image)

    loss = fluid.layers.cross_entropy(pred, label)
    avg_loss = fluid.layers.mean(loss)
    acc = fluid.layers.accuracy(pred, label)

    test_program = fluid.default_main_program().clone(for_test=True)

    adam = fluid.optimizer.Adam(learning_rate=1e-3)
    adam.minimize(avg_loss)

    place = fluid.CPUPlace()
    exe = fluid.Executor(place)
    exe.run(fluid.default_startup_program())
    # fluid.io.load_persistables(exe, '55555')
    compiled_train_prog = fluid.CompiledProgram(fluid.default_main_program())
    compiled_test_prog = fluid.CompiledProgram(test_program)

    for epoch in range(1):
        py_reader.start()
        while True:
            try:
                b_loss, b_acc = exe.run(compiled_train_prog, fetch_list=[avg_loss, acc])
                print('loss: {}, acc: {}'.format(b_loss, b_acc))
            except fluid.core.EOFException:
                py_reader.reset()
                break
        print('-'*30)
        py_reader2.start()
        while True:
            try:
                b_loss, b_acc = exe.run(compiled_test_prog, fetch_list=[avg_loss, acc])
                print('loss: {}, acc: {}'.format(b_loss, b_acc))
            except fluid.core.EOFException:
                py_reader2.reset()
                break

        fluid.io.save_persistables(exe, 'checkpoint')`
